net/dns/manager_windows.go

```diff
@@ -219,22 +219,46 @@ func (m windowsManager) SetDNS(cfg OSConfig) error {
 	if err := m.disableDynamicUpdates(); err != nil {
 		m.logf("disableDynamicUpdates error: %v\n", err)
 	}
-
+	if m.nrptDB == nil && len(cfg.MatchDomains) > 0 {
+		return errors.New("cannot set per-domain resolvers on Windows 7")
+	}
 	if len(cfg.MatchDomains) == 0 {
 		if err := m.setSplitDNS(nil, nil); err != nil {
 			return err
 		}
-		if err := m.setPrimaryDNS(cfg.Nameservers, cfg.SearchDomains); err != nil {
-			return err
-		}
-	} else if m.nrptDB == nil {
-		return errors.New("cannot set per-domain resolvers on Windows 7")
 	} else {
 		if err := m.setSplitDNS(cfg.Nameservers, cfg.MatchDomains); err != nil {
 			return err
 		}
-		// Still set search domains on the interface, since NRPT only
-		// handles query routing and not search domain expansion.
+	}
+
+	isSmartNameResolutionDisabled := func() bool {
+		key, err := m.openKey(`SOFTWARE\Policies\Microsoft\Windows NT\DNSClient`)
+		if err != nil {
+			return false
+		}
+		defer key.Close()
+		val, _, err := key.GetIntegerValue("DisableSmartNameResolution")
+		return err == nil && val == 1
+	}()
+
+	// We still need to set both the search domains and nameservers on the
```

Comment:
Top-level comment is now wrong, we don't always set nameserver. Needs a bit of rewording so the first paragraph only talks about search domains, and the next paragraph talks about when/why we set Nameserver.